core/trino-grammar/src/main/antlr4/io/trino/grammar/sql/SqlBase.g4

```diff
@@ -699,6 +699,7 @@ nullTreatment
 
 string
     : STRING                                #basicStringLiteral
+    | DOLLAR_STRING                         #dollarStringLiteral
```

Comment:
This is the same as [PostgreSQL Dollar-Quoted String Constants](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-DOLLAR-QUOTING), without support for tags. There is no special handling for newlines or indentation -- they are treated literally. 

The same syntax is also supported by the following systems:

* [Snowflake](https://docs.snowflake.com/en/sql-reference/data-types-text#label-dollar-quoted-string-constants)
* [DuckDB](https://duckdb.org/docs/sql/data_types/literal_types.html#dollar-quoted-string-literals)
* [Redshift](https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_PROCEDURE.html)
* [CockroachDB](https://www.cockroachlabs.com/docs/stable/sql-constants#dollar-quoted-string-literals)
* [Vertica](https://docs.vertica.com/24.4.x/en/sql-reference/language-elements/literals/string-literals/dollar-quoted-string-literals/)

[Databricks SQL](https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-create-sql-function.html) seems to support this only for `CREATE FUNCTION` and based on the indentation it also strips indentation.

We could handle indentation in the same or a similar way to Java, possibly using `java.lang.String#stripIndent`, but it might not be worth breaking compatibility with other systems.